framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3772,7 +3772,7 @@ protected void checkMethodInvocability(
             return;
         }
 
-        AnnotatedTypeMirror methodReceiver = method.getReceiverType().getErased();
+        AnnotatedTypeMirror methodReceiver = method.getReceiverType();
         AnnotatedTypeMirror treeReceiver = methodReceiver.shallowCopy(false);
         AnnotatedTypeMirror rcv = atypeFactory.getReceiverType(tree);
```

Comment:
The subtype test below is `typeHierarchy.isSubtype(treeReceiver, methodReceiver);`
`treeReceiver` is a shallow copy of `methodReceiver`, with just the main modifier copied from the actual receiver `rcv` of the method invocation.
So to fix #104 we should really understand better what this code is supposed to do.
What goes from if we use `typeHierarchy.isSubtype(rcv, methodReceiver);`? Why this strange copying of annotations?

Reply:
I will look into #104 after this PR.